tests/lib/eslint/eslint.js

```diff
@@ -1413,297 +1419,391 @@ describe("ESLint", () => {
+							const cwd = getFixturePath(
+								"ts-config-files",
+								"mts",
+								"with-type-commonjs",
+							);
 
-					assert.strictEqual(results.length, 1);
-					assert.strictEqual(results[0].messages.length, 1);
-					assert.strictEqual(results[0].messages[0].severity, 2);
-					assert.strictEqual(
-						results[0].messages[0].ruleId,
-						"no-undef",
-					);
-				});
+							eslint = new ESLint({
+								cwd,
+								flags,
+							});
 
-				it('should load eslint.config.mts config file when we have "type": "module" in nearest `package.json`', async () => {
-					const cwd = getFixturePath(
-						"ts-config-files",
-						"mts",
-						"with-type-module",
-					);
+							const results = await eslint.lintText("foo");
 
-					eslint = new ESLint({
-						cwd,
-						flags,
-					});
+							assert.strictEqual(results.length, 1);
+							assert.strictEqual(results[0].messages.length, 1);
+							assert.strictEqual(
+								results[0].messages[0].severity,
+								2,
+							);
+							assert.strictEqual(
+								results[0].messages[0].ruleId,
+								"no-undef",
+							);
+						});
 
-					const results = await eslint.lintText("foo");
+						it('should load eslint.config.mts config file when we have "type": "module" in nearest `package.json`', async () => {
+							const cwd = getFixturePath(
+								"ts-config-files",
+								"mts",
+								"with-type-module",
+							);
 
-					assert.strictEqual(results.length, 1);
-					assert.strictEqual(results[0].messages.length, 1);
-					assert.strictEqual(results[0].messages[0].severity, 2);
-					assert.strictEqual(
-						results[0].messages[0].ruleId,
-						"no-undef",
-					);
-				});
+							eslint = new ESLint({
+								cwd,
+								flags,
+							});
 
-				it("should find and load eslint.config.cts when present", async () => {
-					const cwd = getFixturePath("ts-config-files", "cts");
+							const results = await eslint.lintText("foo");
 
-					eslint = new ESLint({
-						cwd,
-						flags,
-					});
+							assert.strictEqual(results.length, 1);
+							assert.strictEqual(results[0].messages.length, 1);
+							assert.strictEqual(
+								results[0].messages[0].severity,
+								2,
+							);
+							assert.strictEqual(
+								results[0].messages[0].ruleId,
+								"no-undef",
+							);
+						});
 
-					const results = await eslint.lintText("foo");
+						it("should find and load eslint.config.cts when present", async () => {
+							const cwd = getFixturePath(
+								"ts-config-files",
+								"cts",
+							);
 
-					assert.strictEqual(results.length, 1);
-					assert.strictEqual(results[0].messages.length, 1);
-					assert.strictEqual(results[0].messages[0].severity, 2);
-					assert.strictEqual(
-						results[0].messages[0].ruleId,
-						"no-undef",
-					);
-				});
+							eslint = new ESLint({
+								cwd,
+								flags,
+							});
```

Comment:
To ensure that the config content isn't cached, instead of hard-coding `eslint.config.ts` as a fixture, maybe we could generate it dynamically with some random text inside. For example:

```js
await fsp.writeFile(
	path.join(cwd, "eslint.config.ts"),
	`
	import plugin from "./plugin";

	export default plugin.configs.recommended;

	// Autogenerated on ${Date().toISOString()}`.
);
```

Alternatively, we could rename `eslint.config.ts` to a random name and explicitly pass its path with `overrideConfigFile` in the ESLint options.